### PR TITLE
JAPI-519 DFSClient: BinaryRecordWriter improve float conversions

### PR DESCRIPTION
**Pull request by jpmcmu**
- Added additional Real type conversions

Signed-off-by: James McMullan James.McMullan@lexisnexis.com

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [X] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [X] I have created a corresponding JIRA ticket for this submission
- [X] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [X] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [X] The change has been fully tested:
  - [ ] This change does not cause any existing JUnits to fail.
  - [ ] I have include JUnit coverage to test this change
  - [ ] I have performed system test and covered possible regressions and side effects.
- [X] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] This change fixes the problem, not just the symptom

### CODE REVIEW

**Comment by jpmcmu** [comment]
@rpastrana please review

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordWriter.java [hunk 1]
```diff
@@ -415,10 +415,34 @@ else if (fd.getDataLen() < 8 && fd.getDataLen() > 0)
...
+                }
+                else if (fieldValue instanceof Long)
+                {
+                    value = ((Long) fieldValue).doubleValue();
+                }
+                else if (fieldValue instanceof BigInteger)
+                {
+                    value = ((BigInteger) fieldValue).doubleValue();
+                }
+                else if (fieldValue instanceof BigDecimal)
```
Comment (rpastrana) [inline dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordWriter.java:434]: Does it make sense to warn of possible loss of precision as part of any of these conversions using messages.addMessage("Warning: ... for field: '" + fd.getFieldName() + "'");

**Review by rpastrana — CHANGES_REQUESTED** [review]
@jpmcmu just one comment

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordWriter.java (hunk 1 above)
Comment (jpmcmu) [reply dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordWriter.java:434]: That is a good idea, fixed.

**Review by rpastrana — APPROVED** [review]

**Pull request merged**